library-config/src/lib.rs

```diff
@@ -331,6 +357,8 @@ fn string_operator_match(op: &Operator, matches: &[u8], value: &[u8]) -> bool {
 pub struct LibraryConfig {
     pub name: LibraryConfigName,
     pub value: String,
+    pub source: LibraryConfigSource,
+    pub config_id: Option<String>,
```

Comment:
tbh i'm not sure whether or not we should put it in the vec or not, this may become quite a lot of duplicated data

but i don't really have a better idea today

Reply:
if/when selectors are introduced they can leverage that field too
ex - remote_{version}_{selector_id}